FBKVOController/FBKVOController.m

```diff
@@ -294,13 +315,22 @@ - (void)unobserve:(id)object infos:(NSSet *)infos
   // unregister info
   OSSpinLockLock(&_lock);
   for (_FBKVOInfo *info in infos) {
-    [_infos removeObject:info];
+    if (info->_registered) {
```

Comment:
Same as above method - this will never remove the `info` from the `_infos` table, and will persist it forever.
Instead - we should guard only the `removeObserver` method on the infos.